modules/saml/lib/IdP/SAML2.php

```diff
@@ -238,6 +238,31 @@ private static function getAssertionConsumerService(
         return $spMetadata->getDefaultEndpoint('AssertionConsumerService', $supportedBindings);
     }
 
+    /**
+     * Check a SOAP AuthnRequest.
+     *
+     * @param SimpleSAML_Configuration $idpMetadata The metadata for the IdP.
+     * @param SimpleSAML_Configuration $spMetadata  The metadata for the SP.
+     * @param array                    &$state      The state array.
+     */
+    private static function processSOAPAuthnRequest(SimpleSAML_Configuration $idpMetadata, SimpleSAML_Configuration $spMetadata, SAML2_AuthnRequest $request, array &$state)
+    {
+
+        // Send the response via SOAP.
+        $state['saml:Binding'] = \SAML2\Constants::BINDING_SOAP;
+
+        if (!sspmod_saml_Message::checkSign($spMetadata, $request)) {
+            throw new SimpleSAML_Error_Exception('SOAP authentication request not signed.');
```

Comment:
From reading over [the ECP spec](https://docs.oasis-open.org/security/saml/Post2.0/saml-ecp/v2.0/saml-ecp-v2.0.html) it doesn't seem like an `AuthnRequest` must be signed except in certain conditions...

Particularly, section 2.3.1 states:

> Request for a signed SAML request indicated by the option value "urn:oasis:names:tc:SAML:2.0:profiles:SSO:ecp:2.0:WantAuthnRequestsSigned"

This is stating that the SP should return a signed `AuthnRequest` to the ECP though, not that the IdP wants `AuthnRequest` to be signed. This is further stated in section 2.3.2, which also states:

> In parallel, the service provider MUST include a corresponding `<cb:ChannelBindings>` element in the `<samlp:Extensions>` element of its `<samlp:AuthnRequest>` message for each SOAP header block it attaches, containing channel bindings of a particular type. Within each extension element, the `Type` attribute MAY be set to the channel binding type (if known), and the raw channel binding data MUST be base64-encoded and the result used as the content of the element (per the "default" encoding specified in [ChanBind]). When channel bindings are included, the `<samlp:AuthnRequest>` message MUST be signed via [XMLSig].

However, this is only necessary when channel bindings are used, and is still the responsibility of the SP, not a requirement of the IdP.

In section 2.3.4.1, '"Holder of Key" Subject Confirmation', it is stated that the client must demonstrate possession of the key by signing the entire SOAP message, when applicable, but I think this is a different sort of check than is being performed here.

In section 2.3.6, there are some requirements on the IdP for scenarios where the client is authenticated via a digital signature, but again, that shouldn't require the `AuthnRequest` to be signed.

During testing with O365, Exchange Online _does not_ sign their `AuthnRequest`, and you cannot auth with this check present.